src/test/kotlin/com/makarytskyi/rentcar/repository/CarRepositoryTests.kt

```diff
@@ -0,0 +1,181 @@
+
+    @Test
+    fun `create should insert car and return it with id`() {
+        // GIVEN
+        val unexistingCar = unexistingCar()
+
+        // WHEN
+        val car = carRepository.create(unexistingCar)
+
+        // THEN
+        val foundCar = carRepository.findById(car.id.toString())
+        assertNotNull(foundCar)
+        assertNotNull(car.id)
+    }
+
+    @Test
+    fun `findAll should find all cars`() {
+        // GIVEN
+        val insertedCar1 = carRepository.create(randomCar())
+        val insertedCar2 = carRepository.create(randomCar())
+
+        // WHEN
+        val cars = carRepository.findAll()
+
+        // THEN
+        assertTrue(cars.any { it.brand == insertedCar1.brand && it.plate == insertedCar1.plate })
+        assertTrue(cars.any { it.brand == insertedCar2.brand && it.plate == insertedCar2.plate })
+
+    }
+
+    @Test
+    fun `update should update color of car`() {
+        // GIVEN
+        val car = carRepository.create(randomCar())
+        val updateCar = car.copy(
+            price = null,
+            color = MongoCar.CarColor.BLUE
+        )
+
+        // WHEN
+        carRepository.update(updateCar.id.toString(), updateCar)
+
+        // THEN
+        val updated = carRepository.findById(updateCar.id.toString())
+
+        assertNotNull(updated)
+        assertEquals(updateCar.color, updated?.color)
+    }
+
+    @Test
+    fun `update should update price of car`() {
+        // GIVEN
+        val price = 600
+        val car = carRepository.create(randomCar())
+        val updateCar = car.copy(
+            price = price,
+            color = null
+        )
+
+        // WHEN
+        carRepository.update(updateCar.id.toString(), updateCar)
+
+        // THEN
+        val updated = carRepository.findById(car.id.toString())
+
+        assertNotNull(updated)
+        assertEquals(price, updated?.price)
+        assertEquals(car.color, updated?.color)
+    }
+
+    @Test
+    fun `findByPlate should find existing car by plate`() {
+        // GIVEN
+        val plate = "XX5295YY"
+        val car = randomCar().copy(plate = plate)
+        carRepository.create(car)
+
+        // WHEN
+        val foundCar = carRepository.findByPlate(plate)
+
+        // THEN
+        assertNotNull(foundCar)
+    }
+
+    @Test
+    fun `findByPlate should return null if cant find car by plate`() {
+        // GIVEN
+        val unexistingPlate = "wrongPlate"
+
+        // WHEN
+        val foundCar = carRepository.findByPlate(unexistingPlate)
+
+        // THEN
+        assertNull(foundCar)
+    }
+
+    @Test
+    fun `findAllByBrand should find all cars by brand`() {
+        // GIVEN
+        val brand = "SomeBrand"
+        carRepository.create(randomCar().copy(brand = brand))
+        carRepository.create(randomCar().copy(brand = brand))
+
+        // WHEN
+        val foundCars = carRepository.findAllByBrand(brand)
+
+        // THEN
+        assertEquals(2, foundCars.size)
+    }
```

Comment:
Seems like you are testing save here
What if before your test car(s) with that brand already added?